daffodil-core/src/main/scala/org/apache/daffodil/runtime1/processors/DataProcessor.scala

```diff
@@ -163,12 +163,9 @@ class DataProcessor(
 
   override def withDebugger(dbg: api.debugger.Debugger): DataProcessor = {
     val optDbg = if (dbg eq null) None else Some(dbg)
-    copy(areDebugging = optDbg.isDefined, optDebugger = optDbg)
-  }
-
-  def withDebugging(flag: Boolean): DataProcessor = {
-    val newTunables = tunables.withTunable("allowExternalPathExpressions", flag.toString)
-    copy(areDebugging = flag, tunables = newTunables)
+    val newTunables =
+      tunables.withTunable("allowExternalPathExpressions", optDbg.isDefined.toString)
```

Comment:
I think the previous logic was probably incorrect. If optDbg is not defined, I would suggest we leave `allowExternalPathExpressions` as is and just use the existing tunables as the newTunables. There might be some other use case were a user manually enables this tunable but wants to disable debugging and we shouldn't override that. NOte that we definitely do want to ensure this is enabled if debugging is enabled since it's needed for evaluating debugger expressions.